SimDataFormats/Associations/interface/MuonToTrackingParticleAssociator.h

```diff
@@ -33,15 +35,11 @@ namespace reco {
       impl_->associateMuons(recoToSim, simToReco, muons, type, tpColl);
     }
 
-    void swap(MuonToTrackingParticleAssociator& iOther) {
-      std::swap(impl_, iOther.impl_);
-    }
-
   private:
     MuonToTrackingParticleAssociator( const MuonToTrackingParticleAssociator&) = delete;
     MuonToTrackingParticleAssociator& operator=( const MuonToTrackingParticleAssociator&) = delete;
 
-    MuonToTrackingParticleAssociatorBaseImpl const* impl_;
+    std::unique_ptr<MuonToTrackingParticleAssociatorBaseImpl const> impl_;
```

Comment:
Are these stored objects? If so, you have changed the schema which will stop us from being able to read old ROOT files.

Reply:
No, they are algorithms stored as transient.
https://github.com/cms-sw/cmssw/blob/38a95c4d8839d94350b8a27837f7fac947e1909e/SimDataFormats/Associations/src/classes_def.xml#L2-L7